### PR TITLE
Adding Apigee NAT Address Resource

### PR DESCRIPTION
**Pull request by danistrebel**
Adding Apigee NAT addresses for Apigee Instances as per the [Apigee API](https://cloud.google.com/apigee/docs/reference/apis/apigee/rest/v1/organizations.instances.natAddresses)

fixes https://github.com/hashicorp/terraform-provider-google/issues/9358

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know). => Unable to run the terraform-provider-google tests 😞 .
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_apigee_nat_address`
```

### CODE REVIEW

**Comment by rileykarson** [comment]
Sorry, lost email on my end. @melinath do you mind picking up this review, since you've looked at Apigee more extensively?

File: mmv1/templates/terraform/examples/apigee_nat_address_basic.tf.erb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  authorized_network = google_compute_network.apigee_network.id
+}
+
+resource "google_apigee_instance" "apigee_instance" {
+  name     = "tf-test%{random_suffix}"
+  location = "us-central1-b"
+  org_id   = google_apigee_organization.apigee_org.id
+}
+
+resource "google_apigee_nat_address" "apigee_nat_addres" {
```
Comment (melinath) [inline mmv1/templates/terraform/examples/apigee_nat_address_basic.tf.erb:33]: typo: should be `apigee_nat_adress`

**Review by melinath — COMMENTED** [review]
This seems reasonable. I'll start a /gcbrun.

**Review by melinath — CHANGES_REQUESTED** [review]
It looks like this is failing due to issues in the ansible provider. If you merge master (or rebase your PR) that should address the issue. Could you clarify the issues you had running the acceptance tests for this change?

Sorry that this fell through the cracks. Please feel free to [re-request review](https://github.blog/changelog/2019-02-21-re-request-review-on-a-pull-request/) any time that you're ready for another review.

**Comment by danistrebel** [comment]
thanks @melinath rebased the branch and it looks like we need a /gcbrun again for the remaining check. Thanks

**Comment by melinath** [comment]
Doing a specific CI-run for the new test: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/208683

**Comment by melinath** [comment]
Hm. That failed with the following error:

```
------- Stdout: -------
=== RUN   TestAccApigeeNatAddress_apigeeNatAddressBasicTestExample
=== PAUSE TestAccApigeeNatAddress_apigeeNatAddressBasicTestExample
=== CONT  TestAccApigeeNatAddress_apigeeNatAddressBasicTestExample
provider_test.go:278: Step 1/2 error: Error running apply: exit status 1
2021/10/01 18:48:31 [DEBUG] Using modified User-Agent: Terraform/0.12.30 HashiCorp-terraform-exec/0.13.0
Error: Error waiting to create Instance: Error waiting for Creating Instance: Error code 13, message: Unexpected error during operation. Try again later
on terraform_plugin_test.tf line 56, in resource "google_apigee_instance" "apigee_instance":
56: resource "google_apigee_instance" "apigee_instance" {
--- FAIL: TestAccApigeeNatAddress_apigeeNatAddressBasicTestExample (2443.86s)
FAIL
```

Here's the full request/response:

```
---[ REQUEST ]---------------------------------------
GET /v1/organizations/tf-testvh9ifx91zu/operations/9f6ef1e2-28ba-42eb-b780-a72e4a0f485d?alt=json HTTP/1.1
Host: apigee.googleapis.com
User-Agent: Terraform/0.12.30 (+https://www.terraform.io) Terraform-Plugin-SDK/2.5.0 terraform-provider-google-beta/dev
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------
2021/10/01 19:26:43 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Fri, 01 Oct 2021 19:26:43 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "organizations/tf-testvh9ifx91zu/operations/9f6ef1e2-28ba-42eb-b780-a72e4a0f485d",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.apigee.v1.OperationMetadata",
    "operationType": "INSERT",
    "targetResourceName": "organizations/tf-testvh9ifx91zu/instances/tf-testvh9ifx91zu",
    "state": "FINISHED",
    "progress": {
      "description": "Succeeded",
      "percentDone": 100
    }
  },
  "done": true,
  "error": {
    "code": 13,
    "message": "Unexpected error during operation. Try again later"
  }
}

-----------------------------------------------------
```

Is this probably transient, or more likely flakey? (Starting another CI test run for just this test: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/208699)

**Comment by melinath** [comment]
(FWIW it looks like that last run failed with the same error.)

**Comment by danistrebel** [comment]
@melinath Could you please re-run the cloud build again? I trust the Apigee instance provisioning has stabilized since we last tried. Thanks

**Comment by melinath** [comment]
Running the new test in CI: https://ci-oss.hashicorp.engineering/buildConfiguration/GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstream/232720

**Review by melinath — APPROVED** [review]
LGTM - test failure seems to be related to https://github.com/hashicorp/terraform-provider-google/issues/9988

**Comment by rileykarson** [comment]
Given the basic test for this resource failed on its first nightly, can we revert and resolve the Apigee instance failure prior to merging?

**Comment by melinath** [comment]
@rileykarson sure, if that's preferred.

**Comment by rileykarson** [comment]
A fix-forward also works, I'd like to avoid merging a resource whose sole test is already failing out of the gate is all!

**Comment by melinath** [comment]
given the holidays, I think a rollback is going to be safer / more useful than a fix-forward.

**Pull request merged**